File: client/src/app/types/Message.ts
import { User } from "./User"
import { Workspace } from "./Workspace"

export enum Status {
  READ,
  RECEIVED,
  SENT
}

export interface Message {
  receiverIds: string[] | undefined,
  content: string,
  status : Status | undefined
  senderId: string | undefined,
  workspaceId: string | undefined,
  attachment: Attachment | null,
  createdAt : Date | undefined
  sentAt? : Date
}


export interface Attachment{
  url : string,
  name : string,
  type : string,
  size : number
  extension : string
}

export interface MessageResponse {
  id : string,
  receivers: User[] | undefined,
  content: string,
  status: string ,
  sender: User,
  workspaceId: Workspace | undefined,
  attachment: Attachment | undefined,
  createdAt: Date | undefined
  sentAt?: Date
}


export interface UserStatus{
  username : string,
  isOnline : boolean
}

File: message/src/main/java/com/gather/message/entity/Message.java
package com.gather.message.entity;

import java.util.Collection;
import java.util.Date;
import java.util.UUID;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@Table(name = "message")
public class Message {
  
  @Id
  @GeneratedValue(strategy = GenerationType.UUID)
  private UUID id;
  private Collection<UUID> receiverIds;
  private String content;
  private Status status;
  private UUID senderId;
  private UUID workspaceId;
  private String attachment;
  private Date sentAt;
  private Date createdAt;
}
